### PR TITLE
Refactor convert_jsons_to_xcstrings.py for Improved File Handling and Error Management    

### PR DESCRIPTION
**Pull request by OmarAI2003**
This PR introduces several improvements [discussed here](https://github.com/scribe-org/Scribe-i18n/issues/57) to the `convert_jsons_to_xcstrings.py` script used for converting Scribe-i18n's `Localizable.xcstrings` file to localization JSON files.

### Changes Made
1. **Folder Existence Check:**
   - Added a check to ensure that the "jsons" folder exists before attempting to read files. If the folder does not exist, the script will print an error message and exit gracefully.

2. **Safe File Handling:**
   - Replaced direct file reading with context managers (`with open(...)`) to ensure files are properly closed after being read or written, thus improving resource management.

3. **Loading Language Files:**
   - Pre-loaded all language JSON files into a dictionary at once, reducing the need for repeated file reads during the translation process. This improves performance and clarity.

4. **Improved Key Handling:**
   - Used `dict.get()` for retrieving translations to simplify the logic and improve readability.

### Notes
- Future work could include further modularization and unit testing to enhance the robustness of the codebase.

### CODE REVIEW

**Comment by andrewtavis** [comment]
Could you check this out for a first check as with the last one, @Jag-Marcel? :)

File: Scribe-i18n/scripts/iOS/convert_jsons_to_xcstrings.py [hunk 1]
```diff
@@ -10,35 +10,53 @@
 import os
 
 directory = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
-json_dir_list = os.listdir(os.path.join(directory, "jsons"))
+# Define the path to the "jsons" folder
```
Comment (andrewtavis) [inline Scribe-i18n/scripts/iOS/convert_jsons_to_xcstrings.py:13]: Quick thing, @OmarAI2003: For Pythonic comments if it's its own line then there should be punctuation, so this should have a period.

File: Scribe-i18n/scripts/iOS/convert_jsons_to_xcstrings.py [hunk 2]
```diff
@@ -10,35 +10,53 @@
...
-            open(os.path.join(os.path.join(directory, "jsons"), f"{lang}.json"), "r").read()
-        )
+# Pre-load all JSON files into a dictionary
+lang_data = {}
+for lang in languages:
+    with open(os.path.join(jsons_folder, f"{lang}.json"), "r") as lang_file:
+        lang_data[lang] = json.load(lang_file)
 
-        translation = lang_json[key] if key in lang_json else ""
+for key in base_language_data:          # Iterate over each key in the base language file
```
Comment (andrewtavis) [inline Scribe-i18n/scripts/iOS/convert_jsons_to_xcstrings.py:45]: These commends though don't need punctuation as you have them, but then as they're inline the first letter shouldn't be capitalized. The first word of a comment is only capitalized when it's its own line.

Comment (OmarAI2003) [reply Scribe-i18n/scripts/iOS/convert_jsons_to_xcstrings.py]: > These commends though don't need punctuation as you have them, but then as they're inline the first letter shouldn't be capitalized. The first word of a comment is only capitalized when it's its own line.

I’ve made the edits to the comments as suggested. Thank you for pointing out the details; I’m learning a lot from your feedback!

**Review by andrewtavis — APPROVED** [review]
Very minor changes here @OmarAI2003 :) One thing to note is that we should always do inline comments two spaces after the code that they follow as this is black/ruff code standard. We don't have formatting in this repo as we want to keep it as small as we can and it's just four Python scripts. No stress :)

Thanks for your continued support for these scripts!

**Pull request merged**